esp-hal/src/i2c.rs

```diff
@@ -305,15 +344,14 @@ where
     ///   to indicate writing
     /// - `SR` = repeated start condition
     /// - `SP` = stop condition
-    pub fn transaction(
+    pub fn transaction<'a>(
         &mut self,
         address: u8,
-        operations: &mut [impl I2cOperation],
+        operations: impl Iterator<Item = Operation<'a>>,
```

Comment:
`IntoIterator`? (I think this is needed for slices maybe)
```suggestion
        operations: impl IntoIterator<Item = Operation<'a>>,
```

Reply:
Fair, I'll even add a test for this just to see.

Reply:
Almost, but slices convert into an Iterator that returns references, which is annoying :)